### ISSUE TITLE
Fire Fox profile -  Preference browser.link.open_newwindow may not be overridden: frozen value=2, requested value=3

### ISSUE THREAD

**Issue by AreAllNamesTakenAlready** [SeleniumHQ/selenium#5540]
Selenium Version:  3.9.1

Browser:  Fire Fox

Browser Version:  52

## Expected Behavior -
Hi,

I am trying to load new firefox profile using selenium. I need it for adding firefox extension which is working fine for me. Now second change I  want is open all links in new tab instead of new window. But when I use this profile its opening in new window. I tried to change this using firefox options and profile as well but both are failing with below error - 

Exception in thread "main" java.lang.IllegalArgumentException: Preference browser.link.open_newwindow may not be overridden: frozen value=2, requested value=3

here is my code - 

                FirefoxProfile profile = new FirefoxProfile();
		FirefoxOptions options = new FirefoxOptions();
		DesiredCapabilities capabilitiesDefault = DesiredCapabilities.firefox();
		// capabilitiesDefault.setCapability(CapabilityType.ACCEPT_SSL_CERTS,
		// true);
		// capabilitiesDefault.setCapability("marionette", true);
		capabilitiesDefault.setCapability(FirefoxDriver.PROFILE, profile);
		capabilitiesDefault.setCapability(FirefoxOptions.FIREFOX_OPTIONS,
				options.addPreference("browser.link.open_newwindow", 3));
		// driver = new FirefoxDriver(options);

		driver = new FirefoxDriver(capabilitiesDefault);

interestingly when I comment out the profile code its working fine. That means issue is with creating new profile (or changing the setting for the links).

**Comment by barancev** [comment]
Use new API to init the driver:
```
FirefoxOptions options = new FirefoxOptions();
options.addPreference("browser.link.open_newwindow", 3);
driver = new FirefoxDriver(options);
```

**Comment by jettysoft** [comment]
Hi Barancev,

I'm facing same issue after tried using the following code
FirefoxOptions options = new FirefoxOptions();
options.addPreference("browser.link.open_newwindow", 3);
driver = new FirefoxDriver(options);

with latest version of geckodriver.

Please help me on this.

Thanks

**Comment by azapevalov** [comment]
one moment
what can I do, if I init Remote WebDriver?

**Comment by xiaospider** [comment]
@barancev The problem is I want to `options.addPreference("browser.link.open_newwindow", 3);`
but at the same time I must install a firefox extension by  `profile.addExtension(xpiFile);`
If only set browser.link.open_newwindow and no options.setProfile(profile), it's working. But they can't work both.
I'm using a remote web driver. Is there anyway to use those both?

**Comment by keyantech0** [comment]
issue still works and yet you closed it

**Comment by barancev** [comment]
Actually this issue was fixed in 90dd53c305aab9434190a813558589e284e87dc7 and "defrozed" preferences are available since version 4.0-alpha-3
